dashboard/views.py

```diff
@@ -1,27 +1,30 @@
-from django.forms.models import model_to_dict
-from rules.contrib.views import permission_required, objectgetter
-
-import math, json, logging
+import json
+import logging
+from collections import namedtuple
 from datetime import timedelta
-from django.utils import timezone
+from json import JSONDecodeError
 
+import jsonschema
+import math
 import numpy as np
 import pandas as pd
 from django.conf import settings
 from django.contrib import auth
+from django.core.exceptions import ObjectDoesNotExist
 from django.db import connection as conn
-from django.http import HttpResponse, HttpResponseForbidden
+from django.forms.models import model_to_dict
+from django.http import HttpResponse, HttpResponseForbidden, HttpResponseBadRequest, JsonResponse
 from django.shortcuts import redirect, render
+from django.utils import timezone
 from pinax.eventlog.models import log as eventlog
-from dashboard.event_logs_types.event_logs_types import EventLogTypes
-from dashboard.common.db_util import canvas_id_to_incremented_id
-from dashboard.common import utils
-from django.core.exceptions import ObjectDoesNotExist
-from collections import namedtuple
+from rules.contrib.views import permission_required, objectgetter
 
+from dashboard.common import utils
+from dashboard.common.db_util import canvas_id_to_incremented_id
+from dashboard.event_logs_types.event_logs_types import EventLogTypes
 from dashboard.models import Course, CourseViewOption, Resource, UserDefaultSelection, User
-from dashboard.settings import RESOURCE_VALUES, RESOURCE_VALUES_MAP, RESOURCE_ACCESS_CONFIG
 from dashboard.settings import COURSES_ENABLED
+from dashboard.settings import RESOURCE_VALUES, RESOURCE_VALUES_MAP, RESOURCE_ACCESS_CONFIG
```

Comment:
IntelliJ IDEA has a nice feature to reorganize imports.  It seems to be very good.  It removes imports that were once used, but are no longer, and groups the imports by their origin, like base Python, framework, third-party modules, etc.

It's not necessary, but helpful.